tools/check-copyright.mjs

```diff
@@ -0,0 +1,34 @@
+import { exec } from "child_process";
```

Comment:
This is outstanding. One idea I had (maybe not for this PR) is that we could actually run a theoretical injection of the copyrighted header on changed files, very similar to what we do now. Then, we could run a git command to see if there are modified files as a result of the injection. 

If modified files are detected we'd fail the build.  The added benefit is that now if a contributor runs this locally, they end up with the fixed state ready for staging via git. We do something like this for some of the internal tooling we have for the viya platform and it's worked well I think. We could also autofix, I suppose, but I like still giving the contributor discretion on whether or not the changes get committed / pushed. Thoughts?

Reply:
Yep, this is exactly what I was thinking. As a minimum, we could provide a similar functionality to what lint provides (i.e. we could provide a `--fix` flag to automatically fix the issues for us). I would suggest doing this in another PR, though